src/main/scala/tech/cryptonomic/conseil/tezos/michelson/parser/JsonParser.scala

```diff
@@ -144,21 +149,21 @@ object JsonParser {
       List[Decoder[JsonExpression]](
         Decoder[JsonType].widen,
         Decoder[JsonIntConstant].widen,
-        Decoder[JsonStringConstant].widen,
-        Decoder[JsonEmptyExpression].widen
+        Decoder[JsonStringConstant].widen
+      ).reduceLeft(_ or _)
+
+    val decodeInstructionSequence: Decoder[JsonInstructionSequence] = _.as[List[JsonInstruction]].map(JsonInstructionSequence)
+
+    implicit val decodeInstruction: Decoder[JsonInstruction] =
+      List[Decoder[JsonInstruction]](
+        decodeInstructionSequence.widen,
```

Comment:
I think it wouldn't be terrible to inline this?

Reply:
I wanted to keep it consistent with other ones.

Reply:
oh I didn't realise